extensions/apps/extensions/src/components/pages/extension-release-manager/edit-test-release-page.tsx

```diff
@@ -37,6 +61,84 @@ export const EditTestReleasePage: React.FC<EditTestReleasePageProps> = ({ extens
     });
   }, []);
 
+  const draftReleases = useMemo(() => {
+    try {
+      return JSON.parse(localStorage.getItem(`${DRAFT_RELEASES}-${authenticatedDID}`)) || [];
+    } catch (error) {
+      showErrorNotification(error);
+    }
+  }, [authenticatedDID, showErrorNotification]);
+
+  const localRelease = draftReleases.find(release => release.applicationID === extensionId);
+
+  const draftExtension = getDraftExtension(extensionId, authenticatedDID);
+  const draftExtensionError = draftExtension instanceof Error || false;
+
+  console.log(authenticatedDID, isAuthenticating, draftExtension, draftExtensionError);
+
+  const extensionDataReq = useGetAppsByIdQuery({
+    variables: {
+      id: extensionId,
+    },
+    fetchPolicy: 'cache-first',
+    skip: !authenticatedDID && isAuthenticating,
+  });
+
+  const baseAppInfo = useMemo(() => {
+    if (draftExtension) {
+      return {
+        id: draftExtension.id,
+        name: draftExtension.name,
+        applicationType: draftExtension.applicationType,
+      };
+    }
+
+    if (extensionDataReq.networkStatus === NetworkStatus.ready && extensionDataReq.data) {
+      // select app data
+      return {
+        id: selectAppId(extensionDataReq.data),
+        name: selectAppName(extensionDataReq.data),
+        applicationType: selectApplicationType(extensionDataReq.data),
+      };
+    }
+  }, [draftExtension, extensionDataReq.data, extensionDataReq.networkStatus]);
+
+  useEffect(() => {
+    if (draftExtensionError) {
+      showErrorNotification(draftExtension);
+    }
+  }, [draftExtension, draftExtensionError, showErrorNotification]);
+
+  useEffect(() => {
+    const testModeLoader = getCorePlugins().testModeLoader;
+    let unsubscribe;
+    let timeout;
+    if (testModeLoader) {
+      unsubscribe = testModeLoader.subscribe(({ currentStatus }) => {
+        if (
+          currentStatus &&
+          currentStatus ===
+            getCorePlugins().testModeLoader.getStaticStatusCodes().status
+              .EXTENSION_TEST_LOAD_SUCCESS
+        ) {
+          timeout = setTimeout(() => {
```

Comment:
why is this timeout necessary?

Reply:
the loading is almost instant and it will flash the 'loading modal'. the timeout is required to keep the modal in view for a bit